processing/src/main/java/io/druid/query/search/SearchQueryRunner.java

```diff
@@ -165,7 +188,7 @@ public SearchQueryRunner(Segment segment)
       dimsToSearch = dimensions;
     }
 
-    final Sequence<Cursor> cursors = adapter.makeCursors(filter, segment.getDataInterval(), QueryGranularity.ALL, descending);
+    final Sequence<Cursor> cursors = adapter.makeCursors(filter, interval, query.getGranularity(), descending);
```

Comment:
@navis You're right. This patch only fix the interval bug of search query.
So I changed the title.